pkg/ring/consul_client.go

```diff
@@ -130,14 +130,23 @@ func (p ProtoCodec) Encode(msg interface{}) ([]byte, error) {
 
 // CAS atomically modifies a value in a callback.
 // If value doesn't exist you'll get nil as an argument to your callback.
-func (c *consulClient) CAS(key string, f CASCallback) error {
+func (c *consulClient) CAS(ctx context.Context, key string, f CASCallback) error {
+	return instrument.TimeRequestHistogram(ctx, "CAS loop", consulRequestDuration, func(ctx context.Context) error {
+		return c.cas(ctx, key, f)
+	})
+}
+
+func (c *consulClient) cas(ctx context.Context, key string, f CASCallback) error {
 	var (
 		index   = uint64(0)
 		retries = 10
 		retry   = true
 	)
 	for i := 0; i < retries; i++ {
-		kvp, _, err := c.kv.Get(key, queryOptions)
+		options := &consul.QueryOptions{
```

Comment:
Atleast this should be a consistent read as we are trying to a CAS?

Reply:
My thinking here is that a consistent read before a CAS is complete overkill: if we read an old value, the CAS will fail and we'll try again.  So no point in putting the read through the raft log if we don't need to.